### PR TITLE
docs: added docstrings throughout the package

### PR DESCRIPTION
**Pull request by girish946**
## Summary of change

Added the docstrings generated using [doq](https://pypi.org/project/doq/) to the classes and functions throughout the package.
These are the docstrings generated using numpy docstring template.

As and when the development goes on, the description of these docstrings need to be updated.
This pr finishes the remaining work for [PR93](https://github.com/supertokens/supertokens-python/pull/93). 

## Checklist for important updates

-   [x] Changelog has been updated
-   [ ] `coreDriverInterfaceSupported.json` file has been updated (if needed)
    -   Along with the associated array in `supertokens_python/constants.py`
-   [ ] `frontendDriverInterfaceSupported.json` file has been updated (if needed)
-   [ ] Changes to the version if needed
    -   In `setup.py`
    -   In `supertokens_python/constants.py`
-   [x] Had installed and ran the pre-commit hook
-   [x] Issue this PR against the latest non released version branch.
    -   To know which one it is, run find the latest released tag (`git tag`) in the format `vX.Y.Z`, and then find the latest branch (`git branch --all`) whose `X.Y` is greater than the latest released tag.
    -   If no such branch exists, then create one from the latest released branch.
-   [ ] If have added a new web framework, update the `supertokens_python/utils.py` file to include that in the `FRAMEWORKS` variable
-   [ ] If added a new recipe that has a User type with extra info, then be sure to change the User type in supertokens_python/types.py

### CODE REVIEW

**Comment by rishabhpoddar** [comment]
Thanks! This is going to take a while to review and there are many changes.

File: supertokens_python/__init__.py [hunk 1]
```diff
@@ -30,9 +30,36 @@ def init(app_info: InputAppInfo,
...
+    framework : Literal['fastapi', 'flask', 'django']
+        framework
+    supertokens_config : SupertokensConfig
+        supertokens_config
+    recipe_list : List[Callable[[supertokens.AppInfo], RecipeModule]]
+        recipe_list
+    mode : Union[Literal['asgi', 'wsgi'], None]
+        mode
+    telemetry : Union[bool, None]
+        telemetry
```
Comment (rishabhpoddar) [inline supertokens_python/__init__.py:48]: I really see no point in writting this when the generated docs has these very clearly mentioned as part of the function signature anyway. 

COuld you please explain the benefit of this?

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

File: supertokens_python/__init__.py (hunk 1 above)
Comment (rishabhpoddar) [reply supertokens_python/__init__.py:48]: So in the example of flask that you gave, the function signature is good and there is a lot of description added about each param (and not just the types). In our current API docs (see [this](https://supertokens.com/docs/python/recipe/emailpassword/asyncio/index.html)), you can see that the input and its types are very clearly shown already. So in the docs string, I would expect a good description of the params, and what they do etc.. not just the types (cause you can see the types already).

Am I totally missing the point here?

Comment (girish946) [reply supertokens_python/__init__.py:48]: > So in the docs string, I would expect a good description of the params, and what they do etc.. not just the types (cause you can see the types already).

That is correct.

Also, it won't be possible to add the detailed description immediately to all of the classes and functions' docstrings, this PR just adds the template documentation for each class/method/function which needs to be fixed by someone in the future.

Also to test the document coverage, tools like [docstr-coverage](https://github.com/HunterMcGushion/docstr_coverage) and [interrogate](https://interrogate.readthedocs.io/en/latest/) can be used.

In addition to that, Vscode extensions like [autoDocstring](https://marketplace.visualstudio.com/items?itemName=njpwerner.autodocstring) can be used to generate docstrings during the development.

Comment (rishabhpoddar) [reply supertokens_python/__init__.py:48]: ok thanks! Will have a look.

**Comment by rishabhpoddar** [comment]
Closing this due to inactivity, and cause the added docstrings don't actually explain anything about any of the functions

**Pull request closed without merging**